workflow/artifacts/azureblob/azureblob.go

```diff
@@ -0,0 +1,91 @@
+package azureblob
+
+import (
+	"context"
+	"fmt"
+	"github.com/Azure/azure-storage-blob-go/2017-07-29/azblob"
+	"github.com/argoproj/argo/errors"
+	wfv1 "github.com/argoproj/argo/pkg/apis/workflow/v1alpha1"
+	log "github.com/sirupsen/logrus"
+	"io"
+	"net/url"
+	"os"
+)
+
+// AzureBlobArtifactDriver is a driver for Azure Blob Storage
+type AzureBlobArtifactDriver struct {
+	DefaultEndpointsProtocol string
+	EndpointSuffix           string
+	Container                string
+	Key                      string
+	AccountName              string
+	AccountKey               string
+}
+
+// newMinioClient instantiates a new minio client object.
+func (azblobDriver *AzureBlobArtifactDriver) newAzureBlobClient() (*azblob.ServiceURL, error) {
+	credentials := azblob.NewSharedKeyCredential(azblobDriver.AccountName, azblobDriver.AccountKey)
+	p := azblob.NewPipeline(credentials, azblob.PipelineOptions{})
+	u, err := url.Parse(fmt.Sprintf("https://%s.%s", azblobDriver.AccountName, azblobDriver.EndpointSuffix))
```

Comment:
I think formulating the Azure blob URL from account name and endpoint suffix is limiting. According to this doc, you can customize the URL:
https://docs.microsoft.com/en-us/azure/storage/blobs/storage-custom-domain-name

Following Azure's terminology in `Azure/azure-storage-blob-go`, instead of `EndpointSuffix` can we  use the term `ServiceURL` ? ServiceURL would hold the literal https (or http) URL. and pass it directly to `azblob.NewServiceURL()`